LOGGING/src/NLogTarget/ApplicationInsightsTarget.cs

```diff
@@ -201,25 +193,13 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
 
             try
             {
-                if (this.telemetryClient == null || this.initializationFailed)
-                {
-                    InternalLogger.Debug("ApplicationInsightsTarget.FlushAsync skipped - telemetry client not initialized.");
-                    asyncContinuation(null);
-                    return;
-                }
-
-                InternalLogger.Debug("ApplicationInsightsTarget.FlushAsync flushing telemetry client.");
-                this.TelemetryClient.Flush();
-                if (DateTime.UtcNow.AddSeconds(-30) > this.lastLogEventTime)
+                if (this.telemetryClient == null)
                 {
-                    // Nothing has been written, so nothing to wait for
                     asyncContinuation(null);
                 }
                 else
                 {
-                    // Documentation says it is important to wait after flush, else nothing will happen
-                    // https://docs.microsoft.com/azure/application-insights/app-insights-api-custom-events-metrics#flushing-data
-                    System.Threading.Tasks.Task.Delay(TimeSpan.FromMilliseconds(500)).ContinueWith((task) => asyncContinuation(null));
+                    this.telemetryClient.FlushAsync(System.Threading.CancellationToken.None).ContinueWith(t => asyncContinuation(t.Exception));
```

Comment:
FlushAsync is not ready yet to take this change

Reply:
Can you explain why [TelemetryClient.FlushAsync](https://learn.microsoft.com/dotnet/api/microsoft.applicationinsights.telemetryclient.flushasync) doesn't work and should be avoided? Maybe the API docs should be updated ?

Reply:
Use https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/SHIM/Microsoft.ApplicationInsights.Shim.sln to load the projects

Reply:
If `FlushAsync()` has no valid implementation in the SHIM, then I guess it doesn't make sense to use it. But the console application runs without issue when calling `FlushAsync`.